core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -467,13 +474,25 @@ DeleteFileIndex build() {
 
       return Iterables.transform(
           matchingManifests,
-          manifest ->
-              ManifestFiles.readDeleteManifest(manifest, io, specsById)
-                  .filterRows(dataFilter)
-                  .filterPartitions(partitionFilter)
-                  .caseSensitive(caseSensitive)
-                  .liveEntries()
-      );
+          manifest -> deleteManifestEntries(
```

Comment:
This part becomes quite complicated. Would it make sense to add a method to `ManifestReader` directly? We have one that accepts an expression. We could have one more that accepts `PartitionSet`.

Reply:
Yes refactored and cleaned it up